File: tests/expected/bitops.rs

//! ```cargo
//! [package]
//! edition = "2018"
//! [dependencies]
//!
//! ```

#![allow(clippy::redundant_static_lifetimes)]
#![allow(clippy::upper_case_acronyms)]
#![allow(non_camel_case_types)]
#![allow(non_snake_case)]
#![allow(non_upper_case_globals)]
#![allow(unused_imports)]
#![allow(unused_mut)]
#![allow(unused_parens)]

use std::collections;

pub fn main_func() {
    let mut ands: Vec<bool> = vec![];
    let mut ors: Vec<bool> = vec![];
    let mut xors: Vec<bool> = vec![];
    for a in vec![false, true] {
        for b in vec![false, true] {
            ands.push((a & b));
            ors.push((a | b));
            xors.push((a ^ b));
        }
    }
    assert!(ands == vec![false, false, false, true]);
    assert!(ors == vec![false, true, true, true]);
    assert!(xors == vec![false, true, true, false]);
    println!("{}", "OK");
}

pub fn main() {
    main_func();
}

File: pyjl/tests/tests_py2many-py2many/bitops.jl

function main_func()
    ands::Vector = []
    ors::Vector = []
    xors::Vector = []
    for a in [false, true]
        for b in [false, true]
            push!(ands, (a & b))
            push!(ors, (a | b))
            push!(xors, (a ⊻ b))
        end
    end
    @assert(ands == [false, false, false, true])
    @assert(ors == [false, true, true, true])
    @assert(xors == [false, true, true, false])
    println("OK")
end

function main()
    main_func()
end

main()
